performance_metrics/src/resource_usage_logger.cpp

```diff
@@ -49,15 +49,20 @@ void ResourceUsageLogger::start(std::chrono::milliseconds period)
   std::cout << "[ResourceUsageLogger]: Logging to " << m_filename << std::endl;
 
   m_t1_real_start = std::chrono::steady_clock::now();
-  m_t1_user = std::clock();
-  m_t1_real = std::chrono::steady_clock::now();
+  //m_t1_user = std::clock(); 
+  //m_t1_real = std::chrono::steady_clock::now();
   m_logger_thread_done = false;
 
   // create a detached thread that monitors resource usage periodically
   m_logger_thread = std::thread(
     [ = ]() {
       int64_t i = 1;
       while (m_is_logging) {
+        // Updating m_t1_user and m_t1_real here will have the effect of calculating
+        // resource utilization only over the last `period` milliseconds, *not* 
+        // since program start.
+        m_t1_user = std::clock();
+        m_t1_real = std::chrono::steady_clock::now();
```

Comment:
@mauropasse 

Ok note here how both of these members are updated right before `sleep_until`.  Now the denominator, `time_elapsed_real_ms`, should reflect the `period` duration [here](https://github.com/jfinken/ros2-performance/blob/jfinken/change-avg-ru-calculation/performance_metrics/src/resource_usage_logger.cpp#L128-L131).

Note that I don't think you can compare the actual value from `top` to what you see in the `resources.txt` file.  As I understand it, `top` expresses `%CPU` as a percentage of _total CPU time_.  However, here we are expressing our `cpu%` as the approximate CPU time used by our process over monotonic wall-clock time.  e.g.
```c++
// cpu time of our process   /  monotonic wall clock time, using these
std::clock()                 / std::chrono::steady_clock::now();
```
- https://en.cppreference.com/w/cpp/chrono/c/clock
- https://en.cppreference.com/w/cpp/chrono/steady_clock/now

`top` and our calculation are different.

Reply:
Yes, `top` and our calculation are a bit different. If you multiply the CPU% we get in our way, by the number of cores of your system, the values are quite similar.
I mostly refered to the `top` approach of averaging every 1 or 2 seconds, so it doesn't reflect the past before that.
For that, the CPU load is what helps. I don't think we care about that so our approach is fine!